pyiron_base/project/generic.py

```diff
@@ -1730,3 +1712,114 @@ def table(self, job_name="table", delete_existing_job=False):
         )
         table.analysis_project = self._project
         return table
+
+
+class _JobByAttribute(ABC):
```

Comment:
Does it make sense to move the new classes to a separate python module? With over 1700 lines the project module is already very lengthy.

Reply:
Yep, good call.